packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts

```diff
@@ -878,6 +1123,15 @@ export class GaussianSplattingMesh extends Mesh {
         return runCoroutineAsync(GaussianSplattingMesh.ConvertPLYToSplat(data, true), createYieldingScheduler());
     }
 
+    /**
+     * Converts a .ply with SH data array buffer to splat
+     * if data array buffer is not ply, returns the original buffer
+     * @param data the .ply data to load
+     * @returns the loaded splat buffer with SH
+     */
+    public static async ConvertPLYWithSHToSplatAsync(data: ArrayBuffer) {
+        return runCoroutineAsync(GaussianSplattingMesh.ConvertPLYWithSHToSplat(data, true), createYieldingScheduler());

```

Comment:
> just wondering (might have missed it) where is the coroutine used?

@RaananW here

Reply:
Ho! I've missed it in [/splatFileLoader.ts](https://github.com/BabylonJS/Babylon.js/pull/16055/files#diff-e0ea2d460e0a730fe5f692ccbae665bea3ba34559aa0449fc26b1f8b203b702d)